docs(chatbot): add component pages (#244)
siemens/ix-docs: docs/components/user-message/guide.md

---
doc-type: 'tab-item'
description: 'Usage guide for user message'
---

# User message - Usage

> User messages display a single prompt submitted by users in a conversational thread. We recommend using them to preserve what users asked, attached or edited so follow-up answers keep the right context.

Reviewer: ```suggestion
User messages display a single prompt submitted by users in a conversational thread. We recommend using them to preserve what users asked, attached or edited so follow-up answers maintain context.
```

![User message anatomy](https://www.figma.com/design/wEptRgAezDU1z80Cn3eZ0o/iX-Documentation-illustrations?node-id=7959-1370&t=8Rj3ErabF16Vm3lH-4)

1. Attachments
2. Message
3. Actions

## Options

> - **Attachments:** If users submit files together with a prompt, show those [attachments](../chat-attachment) directly with the message so the request stays understandable later in the thread.

Reviewer: ```suggestion
- **Attachments:** When users attach files with their prompts, show those [attachments](../chat-attachment) with the message to maintain context even after multiple turns in the conversation. 
```

- **Message:** Show the original user input as the main message content without alterations.
- **Actions:** Add only the few actions users need for their own prompt, e.g. copy, edit or open a compact overflow menu. We recommend using subtle tertiary [icon buttons](../icon-button) so actions stay secondary to the message.

## Behavior in context

> - **Interaction:** User messages keep the submitted prompt visible as the main record of what users asked.

Reviewer: ```suggestion
- **Interaction:** User messages keep the sent prompts visible as a chat history of user input.
```

> - **Actions:** Message actions are only shown when users hover over the message with a mouse, tap the message on touch devices or reach the message with the `Tab` key.

Reviewer: the hover text and action labels are available here: https://www.figma.com/design/lqjt9c5IzzwQ4eJ4nqG7Kv/AI-Terminology?node-id=1-9&t=8g9VIGSar5B6wwtC-1

Reviewer 2: Technically, there's no "hover text". Is it an aria-label? A tooltip? Title? - in every case, we need to add these recommendations to the options section

Also, is this available open source (=text recommendations can be added to this markdown) or inner source (in gitlab)? - in this Figma, it's really difficult to find them

Reviewer 2: ```suggestion
- **Actions:** Message actions are only shown when users hover over the message with a mouse, tap the message on touch devices or reach the message with the `Tab` key (for Siemens AG see the [AI UX terminology guide](https://www.figma.com/design/lqjt9c5IzzwQ4eJ4nqG7Kv/AI-Terminology?node-id=1-9&t=8g9VIGSar5B6wwtC-1) on labelling actions).
```

> - **Actions:** Message actions are only shown when users hover over the message with a mouse, tap the message on touch devices or reach the message with the `Tab` key (for Siemens AG see the [AI UX terminology guide](https://www.figma.com/design/lqjt9c5IzzwQ4eJ4nqG7Kv/AI-Terminology?node-id=1-9&t=8g9VIGSar5B6wwtC-1) on labelling actions).

Reviewer 3: Users need to request access to view the AI Terminology file. If we link resources here, they should be accessible for Siemens Figma users without requesting access.

Reviewer 2: @tokyojen you suggested to adding the file, what's your take?

- **Placement:** User messages are always placed on the right side of the [chat](../chat/) to visually distinguish them from [AI messages](../ai-message) on the left side.

> - **Responsiveness:** User messages take from 45 to 80 % of the chat's container width, depending on the viewport width.

Reviewer: ```suggestion
- **Responsiveness:** User messages take from 45 to 80% of the chat's container width, depending on the viewport width.
```

## Dos and Don’ts

> - Do offer only the few actions users need after sending, e.g. copy, edit or more actions

Reviewer: ```suggestion
- Do offer only the few actions users need after sending, e.g. copy or edit
```

> - Do keep the original message wording and attachments visible so users can review what they submitted

Reviewer: ```suggestion
- Do keep the messages and attachments visible as a continuous chat
```

> 

Reviewer: Maybe an idea for a dont: 

"Don't add limits to how many messages are visible, instead always display full chat sessions"

Reviewer 2: developers will love that 😂 I'll add it anyways
